specs/requests/teams_spec.py

```diff
@@ -0,0 +1,129 @@
+            expect(self.response.json["teams"]).to(have_len(2))
+            expect(self.response.json["teams"]).to(contain_only(
+                have_key("name", "Blue"),
+                have_key("name", "Red")
+            ))
+
+    with description('POST'):
+        with description('valid request'):
+            with before.each:
+                self.response = self.context.client().post(
+                    "/teams",
+                    data={"name": "My Team"})
+
+            with it('responds with 200'):
+                expect(self.response.status_code).to(equal(200))
+
+            with it('responds with correct team'):
+                expect(self.response.json).to(have_key("name", "My Team"))
+                expect(self.response.json).to(have_key("id"))
+
+            with it('saves the team'):
+                expect(Team.exists(id_=self.response.json["id"], name="My Name")).to(be(True))
+
+        with description('invalid request'):
+            with before.each:
+                self.response = self.context.client().post(
+                    "/teams",
+                    data={"name": ""})
+
+            with it('declines empty name'):
+                expect(self.response.status_code).to(equal(422))
+
+            with it('does not save invalid teams'):
+                expect(Team.all()).to(be_empty)
+
+    with description('/:id'):
+        with description('GET'):
+            with description('valid id'):
+                with before.each:
+                    self.team = Team.create(name='Blue')
+                    self.response = self.context.client().get(f"/teams/{self.team.id_}")
+
+                with it('responds with 200'):
+                    expect(self.response.status_code).to(equal(200))
+
+                with it('responds with the correct team'):
+                    expect(self.response.json).to(have_key("name", "Blue"))
+                    expect(self.response.json).to(have_key("id", self.team.id_.urn[9:]))
+
+            with description('invalid id'):
+                with before.each:
+                    team = Team.create(name='Blue')
+                    team.delete()
+                    self.response = self.context.client().get(f"/teams/{team.id_}")
+
+                with it('responds error 404'):
+                    expect(self.response.status_code).to(equal(404))
+
+        with description('PUT'):
+            with description('valid id'):
+                with before.each:
+                    self.team = Team.create(name='Blue')
+                    self.response = self.context.client().put(
+                        f"/teams/{self.team.id_}",
+                        data={"name": "Red"})
+
+                with it('responds with 200'):
+                    expect(self.response.status_code).to(equal(200))
+
+                with it('updates the team correctly'):
+                    fresh_team = Team.find_by(id_=self.team.id_)
```

Comment:
this line should be in the before block. Or if you don't want to execute it every time because it is just needed once create another description block: one for the response and one fore the updated team. Because you aren't using the `self.response` in this statement either.